### ISSUE TITLE
Make color painting behave like Super Slicer's enhancement to Prusa - painted colors aren't just one surface layer

### ISSUE THREAD

**Issue by sprior** [bambulab/BambuStudio#958]
I tried Bambu Studio's color painting feature for a print and found that the painted color is only one layer/perimeter deep which when printed allows the underneath color to affect the saturation of the surface color.  I would suggest that this is rarely what the designer wants.
![2022-12-26 22 29 19](https://user-images.githubusercontent.com/3361800/209610766-8c8b62fe-30d5-46d4-8fd1-545398777aaf.jpg)

**Describe the solution you'd like**
I tried this in Prusa Slicer and found it behaves the same as Bambu Slicer - the painted color is only on the surface.  I then tried it in Super Slicer and found that the painted color is applied to the full depth of the feature (see screen shot).

I would suggest switching to Super Slicer's implementation of color painting.

**Describe alternatives you've considered**
There appears to be no way to color the model in Super Slicer and export the model 3mf file to Bambu Studio because it would only be in the project 3mf file.  The only other alternative would be if the user has the original model and can export it as a multi part body but this isn't available if you're starting from a model you didn't design.

**Additional context**
![Screenshot 2022-12-26 230140](https://user-images.githubusercontent.com/3361800/209610673-65de7939-e05d-46b8-a9f8-0f1ed8adb893.png)

![Screenshot 2022-12-26 225654](https://user-images.githubusercontent.com/3361800/209610683-4fb1f8eb-e25e-4a83-b209-b64bb61fc1fe.png)

**Comment by randyre** [comment]
I actually use the single layer a lot as it reduces filament switching, but not for every color. What I would like to see is both options available as I have also seen where some colors could use more layers or a particular model benefits from a different way of painting. 

maybe a box in the coloring tool to allow us to input how many layers the painting will affect ort something like that.

**Comment by CameronBigger** [comment]
@randyre I think your idea of adding a "layer paint depth" field like you describe would make everyone happy. It could default to '1' and work like it currently does, but would allow pretty easy customization of how deep it should go for when surface painting isn't enough.

**Comment by medic6578** [comment]
I have run into this myself. What would be nice would be allowing me to choose how many layers deep I want the color to go. So a couple of check boxes for 1 layer or all layers and a slider to choose the layer depth if I want to go that route.

**Comment by gitjpk** [comment]
+1 on this issue. Please allow us to decide the number of layers

**Comment by Ing-Dom** [comment]
this would be a great feature! +1 !

**Comment by woodenforest** [comment]
+1 i really need this feature!

**Comment by randyre** [comment]
I agree. Here is a picture with the standard painting. Really needed a couple more layers to make it opaque.
![Photo Apr 09 2023, 10 45 36](https://user-images.githubusercontent.com/60475095/231585194-249db3ea-18bf-406e-804b-ca7caab100a7.jpg)

**Comment by Britford** [comment]
+1 This feature would make the AMS much more useful.
